app/controllers/concerns/application_shared.rb

```diff
@@ -15,4 +15,35 @@ def set_timezone
     # Reset timezone for the next thread
     Time.zone = default_timezone
   end
-end
\ No newline at end of file
+
+  private
+
+  # Get the allowed parameters from the Apipie documentation in the API controllers.
+  def safe_params
+    apipie_params = Apipie.app.get_method_description("v2##{controller_name}", action_name).to_json[:params]
+    model_api_description = apipie_params.find { |param| param[:name] == controller_name.singularize }
+    params.require(controller_name.singularize.to_sym).
+      permit(*safe_attributes_from_api_description(model_api_description))
+  end
+
+  def safe_attributes_from_api_description(model_api_description)
+    model_api_description[:params].map do |param|
+      if param_group?(param)
+        { param[:name].to_sym => build_safe_params(param[:params]) }
```

Comment:
Apologies for the name, it's a bit unclear but this parameters come basically from Apipie descriptions, I realize now the naming makes it look as if they were actual user params. I'll update the PR with better variable naming.